src/main/java/spoon/support/reflect/code/CtCommentImpl.java

```diff
@@ -63,7 +65,7 @@ public String getRawContent() {
 	@Override
 	public <E extends CtComment> E setContent(String content) {
 		getFactory().getEnvironment().getModelChangeListener().onObjectUpdate(this, CtRole.COMMENT_CONTENT, content, this.content);
-		this.content = content;
+		this.content = cleanComment(content);
```

Comment:
Shouldn't model change listener receive cleaned comment? ... because it is the real new value
I am not sure. 
WDYT?

Reply:
I thought about that too. I'm not sure either. The problem is that if we add an assignment as first statement, this will break the test verifying that the first statement calls the listener .

Reply:
so let's keep it simple like it is now. May be just put there a comment that we are not sure and it is possible to move it up when somebody founds it useful later.

Reply:
Agree. I mark this point as resolved.